pkg/apis/extensions/types.go

```diff
@@ -87,6 +88,18 @@ type CPUTargetUtilization struct {
 	TargetPercentage int `json:"targetPercentage"`
 }
 
+// Alpha-level support for Custom Metrics in HPA (as annotations).
+type CustomMetricUtilization struct {
+	// Custom Metric name.
+	Name string `json:"name"`
+	// Custom Metric value (average).
+	Value resource.Quantity `json:"value"`
```

Comment:
I think this type should be as close as possible to what you plan to move into the HPA object.  It sounds like the fields in Spec and Status should have different names, so you probably need two objects?